tests/ui/consts/const-eval/raw-pointer-ub.rs

```diff
@@ -0,0 +1,11 @@
+#![feature(const_pointer_byte_offsets)]
+#![feature(pointer_byte_offsets)]
+
+const MISALIGNED: () = unsafe {
+    let mem = [0u32; 8];
+    let ptr = mem.as_ptr().byte_add(1);
+    let _val = *ptr; //~ERROR: evaluation of constant value failed
+    //~^NOTE: accessing memory with alignment 1, but alignment 4 is required
+};
```

Comment:
Strangely this test is already a hard error on nightly... https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=24a9a0f6ecadb7064178b092b4fab1a6

Reply:
Ah, here's why:

https://github.com/rust-lang/rust/blob/3ec4b3bc8c18fc217cd8d90426024f1ec1fd4a6a/compiler/rustc_const_eval/src/interpret/place.rs#L474-L475

`check_ptr_access_align` will always check alignment, so this entirely circumvented the future-incompat machinery.